packages/core/utils/src/dml/entity.ts

```diff
@@ -16,8 +19,34 @@ import { BelongsTo } from "./relations/belongs-to"
 export class DmlEntity<Schema extends DMLSchema> implements IDmlEntity<Schema> {
   [IsDmlEntity]: true = true
 
+  name: string
+
+  readonly #tableName: string
   #cascades: EntityCascades<string[]> = {}
-  constructor(public name: string, public schema: Schema) {}
+
+  constructor(nameOrConfig: Config, public schema: Schema) {
+    if (isString(nameOrConfig)) {
```

Comment:
NIT: Maybe create a helper function `const {name, tableName} = getNameAndTable(nameOrConfig)` so we don't have so much logic in the constructor.